src/Components/Assets/AssetFilter.tsx

```diff
@@ -38,12 +32,32 @@ function AssetFilter(props: any) {
   const [asset_class, setAssetClass] = useState<string>(
     filter.asset_class || ""
   );
-  const [facilityId, setFacilityId] = useState<number | "">(filter.facility);
+  const [facilityId, setFacilityId] = useState<string | "">(filter.facility);
   const [locationId, setLocationId] = useState<string | "">(filter.location);
   const [qParams, _] = useQueryParams();
 
+  const { data: facilityData } = useQuery(routes.getPermittedFacility, {
+    pathParams: { id: facilityId },
+  });
+
+  const { data: locationData } = useQuery(routes.getFacilityAssetLocation, {
+    pathParams: {
+      facilityId: String(facilityId),
+      locationId: String(locationId),
+    },
+  });
```

Comment:
You can wait for this query to be dispatched until related queries are satisfied using this approach.

```suggestion
  const { data: locationData } = useQuery(routes.getFacilityAssetLocation, {
    pathParams: {
      facilityId: String(facilityId),
      locationId: String(locationId),
    },
    prefetch: !!(facilityId && locationId)
  });
```